src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -0,0 +1,324 @@
+                    ["asc"] = "true",
+                });
+
+            return InvokeSonarQubeApi(apiPath, token,
+                stringResponse => JObject.Parse(stringResponse)["components"].ToObject<ComponentResponse[]>());
+        }
+
+        public Task<Result<ServerIssue[]>> GetIssuesAsync(string key, CancellationToken token)
+        {
+            // Since 5.1; internal
+            var apiPath = BuildRelativeUrl("batch/issues",
+                new Dictionary<string, string>
+                {
+                    ["key"] = key
+                });
+
+            return InvokeSonarQubeApi(apiPath, token,
+                async (HttpResponseMessage response) =>
+                {
+                    var byteArray = await response.Content.ReadAsByteArrayAsync();
+                    // Protobuf for C# throws when trying to read outside of the buffer and ReadAsStreamAsync returns a non
+                    // seekable stream so we can't determine when to stop. The hack is to use an intermediate MemoryStream
+                    // so we can control when to stop reading.
+                    // Note we might want to use FileStream instead to avoid intensive memory usage.
+                    using (var stream = new MemoryStream(byteArray))
+                    {
+                        return ReadFromProtobufStream(stream, ServerIssue.Parser).ToArray();
+                    }
+                });
+        }
+
+        public Task<Result<OrganizationResponse[]>> GetOrganizationsAsync(OrganizationRequest request, CancellationToken token)
+        {
+            // Since 6.2; internal
+            var apiPath = BuildRelativeUrl("api/organizations/search",
+                new Dictionary<string, string>
+                {
+                    ["p"] = request.Page.ToString(),
+                    ["ps"] = request.PageSize.ToString()
+                });
+
+            return InvokeSonarQubeApi(apiPath, token,
+                stringResponse => JObject.Parse(stringResponse)["organizations"].ToObject<OrganizationResponse[]>());
+        }
+
+        public Task<Result<PluginResponse[]>> GetPluginsAsync(CancellationToken token)
+        {
+            // Since 2.10; internal
+            var serverPluginsInstalledApiUrl = BuildRelativeUrl("api/updatecenter/installed_plugins");
+
+            return InvokeSonarQubeApi(serverPluginsInstalledApiUrl, token,
+                stringResponse => ProcessJsonResponse<PluginResponse[]>(stringResponse, token));
+        }
+
+        public Task<Result<ProjectResponse[]>> GetProjectsAsync(CancellationToken token)
+        {
+            // Since 2.10
+            var projectsApiUrl = BuildRelativeUrl("api/projects/index");
+
+            return InvokeSonarQubeApi(projectsApiUrl, token,
+                stringResponse => ProcessJsonResponse<ProjectResponse[]>(stringResponse, token));
+        }
+
+        public Task<Result<PropertyResponse[]>> GetPropertiesAsync(CancellationToken token)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)